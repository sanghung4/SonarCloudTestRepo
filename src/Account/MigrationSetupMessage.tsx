import {
  Box,
  Grid,
  Typography,
  useScreenSize,
  Button,
  useSnackbar,
  CircularProgress
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';

import { useResendLegacyInviteEmailMutation } from 'generated/graphql';
import { MaxIcon, WelcomeToMaxIcon } from 'icons';
import Link from 'common/Link';
import useDocumentTitle from 'hooks/useDocumentTitle';

type LocationState = {
  email: string;
};

function MigrationSetupMessage() {
  /**
   * Custom Hooks
   */
  const location = useLocation<LocationState>();
  const { isSmallScreen } = useScreenSize();
  const { pushAlert } = useSnackbar();
  const { t } = useTranslation();
  useDocumentTitle(t('common.welcomeToMaX'));

  /**
   * GraphQL Queries
   */
  const [sendInviteEmail, { loading }] = useResendLegacyInviteEmailMutation({
    onCompleted: () =>
      pushAlert(t('register.emailSentNotification'), { variant: 'success' }),
    onError: () =>
      pushAlert(t('register.emailSentErrorNotification'), { variant: 'error' })
  });

  return (
    <Grid
      container
      direction="column"
      sx={{ alignItems: 'center', pb: isSmallScreen ? 6 : 0, px: 6, mt: 8 }}
    >
      <Box display="flex" flexDirection="row">
        <Typography variant="h3" color="primary" sx={{ fontWeight: 700 }}>
          {t('account.welcomeTo')}
        </Typography>
        <Box
          sx={{
            pt: 0.125,
            pl: 0.125
          }}
        >
          <MaxIcon height={25} width={75} />
        </Box>
      </Box>

      <Grid item sx={{ p: isSmallScreen ? 2 : 4, pb: isSmallScreen ? 2 : 4 }}>
        <WelcomeToMaxIcon />
      </Grid>
      <Grid
        item
        sx={{
          textAlign: 'center',
          pt: 2,
          maxWidth: isSmallScreen ? 640 : '30vw'
        }}
      >
        <Typography>
          {`${t('account.maxWelcomeLine1')} ${t('account.maxWelcomeLine2')}`}
        </Typography>
      </Grid>
      {location.state ? (
        <>
          <Grid item>
            <Box pt={6}>
              <Typography>{t('register.noEmail')}</Typography>
            </Box>
          </Grid>
          <Grid item>
            <Button
              disabled={loading}
              fullWidth
              variant="text"
              color="primaryLight"
              size="medium"
              data-testid="resend-email-button"
              onClick={() =>
                sendInviteEmail({
                  variables: { legacyUserEmail: location.state.email }
                })
              }
              sx={{ p: 0 }}
            >
              {loading ? (
                <CircularProgress size={15} />
              ) : (
                t('register.resendEmail')
              )}
            </Button>
          </Grid>
          <Grid item container justifyContent="center">
            <Box mt={8} mb={5} maxWidth={450}>
              <Typography
                color="primary"
                variant="h4"
                align="center"
                sx={{ fontWeight: 500, fontSize: 20, pb: 1 }}
              >
                {t('common.havingTrouble')}
              </Typography>
              <Typography
                align="center"
                sx={{ fontWeight: 400, lineHeight: '24px' }}
              >
                {t('common.hereToHelp')}
                <Link to="/location-search">{t('home.supportUrl1')}</Link>
                {t('home.supportSub2')}
                <Link to="/support">{t('home.supportUrl2')}</Link>
              </Typography>
            </Box>
          </Grid>
        </>
      ) : null}
    </Grid>
  );
}

export default MigrationSetupMessage;
