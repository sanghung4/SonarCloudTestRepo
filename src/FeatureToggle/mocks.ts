import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
