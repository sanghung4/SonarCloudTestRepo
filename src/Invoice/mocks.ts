import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
