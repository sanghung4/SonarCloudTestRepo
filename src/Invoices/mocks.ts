import { subMonths } from 'date-fns';
import { Invoice, InvoicesDocument } from 'generated/graphql';
import { mockRow } from 'test-utils/mockTableInstance';
import { formatDate } from 'utils/dates';

export const success = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'ECLIPSE'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Current',
            customerPo: '1096872',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const successMincron = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'MINCRON'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '6566',
            jobName: 'NEP 7 ASHVELLE'
          },
          {
            age: 'Current',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '56787',
            jobName: 'WALTER HORN'
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'SHOP',
            jobName: 'SEQUAYESH TR'
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'RETREAT',
            jobName: 'RETREAT AT ARDEN'
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'AVLNEP7',
            jobName: 'WALTER HORN'
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'AIKENON',
            jobName: 'AIKEN AVL'
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const mockTestTableMemo = {
  invoices: {
    bucketNinety: 0.0,
    bucketOneTwenty: 0.0,
    bucketSixty: 0.0,
    bucketThirty: 0.0,
    bucketFuture: 0.0,
    currentAmt: 35224.96,
    invoices: [
      {
        age: 'Future',
        customerPo: '109687',
        invoiceDate: '05/02/2022',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227972.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '6566',
        jobName: 'NEP 7 ASHVELLE'
      },
      {
        age: 'Current',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227975.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '56787',
        jobName: 'WALTER HORN'
      },
      {
        age: 'Over120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220294.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'SHOP',
        jobName: 'SEQUAYESH TR'
      },
      {
        age: '91-120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110222345.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'RETREAT',
        jobName: 'RETREAT AT ARDEN'
      },
      {
        age: 'Deposit',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220974.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AVLNEP7',
        jobName: 'WALTER HORN'
      },
      {
        age: '31-60',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227973.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AIKENON',
        jobName: 'AIKEN AVL'
      }
    ],
    totalAmt: 35232.27,
    totalPastDue: 7.3085938
  }
};

const rowToMock = success.result.data.invoices.invoices[0];
export const mockInvoiceRow = mockRow<Invoice>(rowToMock, rowToMock, {});
