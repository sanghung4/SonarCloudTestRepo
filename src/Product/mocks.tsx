import {
  GetProductDocument,
  GetProductPricingDocument
} from 'generated/graphql';

export const success = {
  request: {
    query: GetProductDocument,
    variables: {
      productInput: { productId: 'MSC-33860', customerNumber: undefined }
    }
  },
  result: {
    data: {
      product: {
        id: 'MSC-33860',
        partNumber: '33860',
        name: '1/2" Wrot Copper Cap',
        manufacturerName: 'Generic',
        manufacturerNumber: 'CCAPD',
        price: 0,
        stock: null,
        taxonomy: null,
        categories: [
          'Pipe & Fittings',
          'Metal Pipe & Fittings',
          'Copper Tube & Fittings'
        ],
        technicalDocuments: [
          {
            name: 'MFR Catalog Page',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100145/MUELERE00056_18_20.pdf',
            __typename: 'TechDoc'
          },
          {
            name: 'Technical Specification',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100240/MUELERE03088_1.pdf',
            __typename: 'TechDoc'
          },
          {
            name: 'MFR Item Data',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100145/MUELERE00601_1.pdf',
            __typename: 'TechDoc'
          }
        ],
        environmentalOptions: [],
        upc: '68576823241',
        unspsc: '40183104',
        seriesModelFigureNumber: null,
        productOverview:
          'Cap Fitting; Type Round Head; Nominal Size 1/2"; End Connection Female Soldered; Material Copper; Process Wrot; Application Ground Potable Water Supply System; Applicable Standard ASME B16.22, ANSI/NSF 61 G; Packaging Quantity 50 per Inner Pack, 1000 per Master Carton',
        featuresAndBenefits:
          'Copper Solder-Joint Fittings for Supply/Pressurized Systems have Been the Leading Brand of Copper Fittings for Over 80 Years; Available in Both Wrot Copper and Cast Bronze; The Acknowledged Experts at Engineering and Manufacturing Precision Solder-Joint Copper Fittings; Quality, Consistency and Reliability have Made the Streamline Brand Trusted and Specified All Around the World',
        techSpecifications: [
          {
            name: 'material',
            value: 'Copper',
            __typename: 'TechSpec'
          },
          {
            name: 'Applicable Standard',
            value: 'ASME B16.22, ANSI/NSF 61 G',
            __typename: 'TechSpec'
          },
          {
            name: 'Application',
            value: 'Ground Potable Water Supply System',
            __typename: 'TechSpec'
          },
          {
            name: 'End Connection',
            value: 'Female Soldered',
            __typename: 'TechSpec'
          },
          {
            name: 'Material',
            value: 'Copper',
            __typename: 'TechSpec'
          },
          {
            name: 'Nominal Size',
            value: '1/2"',
            __typename: 'TechSpec'
          },
          {
            name: 'Process',
            value: 'Wrot',
            __typename: 'TechSpec'
          },
          {
            name: 'Type',
            value: 'Round Head',
            __typename: 'TechSpec'
          }
        ],
        imageUrls: {
          thumb:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_SML.jpg',
          small:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_SML.jpg',
          medium:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_MED.jpg',
          large:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_LRG.jpg',
          __typename: 'ImageUrls'
        },
        packageDimensions: null,
        minIncrementQty: 0,
        erp: 'ECLIPSE',
        __typename: 'Product'
      }
    }
  }
};

export const pricing = {
  request: {
    query: GetProductPricingDocument,
    variables: {
      input: {
        customerId: '',
        branchId: 'any',
        productIds: ['33860'],
        includeListData: false
      }
    }
  },
  result: {
    data: {
      productPricing: {
        customerId: 'testAccount',
        branch: 'any',
        products: [
          {
            productId: '33860',
            orderUom: 'ea',
            catalogId: '',
            branchAvailableQty: 2,
            totalAvailableQty: 3,
            listIds: [],
            sellPrice: 3.256
          }
        ]
      }
    }
  }
};
