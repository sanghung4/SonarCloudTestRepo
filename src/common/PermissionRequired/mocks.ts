import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
