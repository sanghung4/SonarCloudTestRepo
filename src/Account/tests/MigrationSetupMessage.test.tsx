import { SnackbarProvider } from '@dialexa/reece-component-library';
import { act, fireEvent } from '@testing-library/react';

import MigrationSetupMessage from 'Account/MigrationSetupMessage';
import { migrationMocks } from 'Account/tests/mocks';
import * as t from 'locales/en/translation.json';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

const locationMock: { state?: { email: string } } = {
  state: { email: '' }
};
const mockEmail = migrationMocks[0].request.variables.legacyUserEmail;

jest.mock('react-router-dom', () => ({
  useLocation: () => locationMock
}));

describe('Migration Setup Message', () => {
  it('should match snapshot in desktop', () => {
    setBreakpoint('desktop');
    locationMock.state = { email: mockEmail };
    const { container } = render(<MigrationSetupMessage />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot in desktop when location state is undefined', () => {
    setBreakpoint('desktop');
    locationMock.state = undefined;
    const { container } = render(<MigrationSetupMessage />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot in mobile', () => {
    setBreakpoint('mobile');
    locationMock.state = { email: mockEmail };
    const { container } = render(<MigrationSetupMessage />);
    expect(container).toMatchSnapshot();
  });

  it('failed to send invite email', async () => {
    setBreakpoint('desktop');
    locationMock.state = { email: mockEmail };
    const { getByTestId, findByText } = render(
      <SnackbarProvider>
        <MigrationSetupMessage />
      </SnackbarProvider>
    );

    act(() => {
      fireEvent.click(getByTestId('resend-email-button'));
    });

    const errorMsg = await findByText(t.register.emailSentErrorNotification);
    expect(errorMsg).toBeInTheDocument();
  });

  it('successfully to send invite email', async () => {
    setBreakpoint('desktop');
    locationMock.state = { email: mockEmail };
    const { getByTestId, findByText } = render(
      <SnackbarProvider>
        <MigrationSetupMessage />
      </SnackbarProvider>,
      { mocks: migrationMocks }
    );

    act(() => {
      fireEvent.click(getByTestId('resend-email-button'));
    });

    const successMsg = await findByText(t.register.emailSentNotification);
    expect(successMsg).toBeInTheDocument();
  });
});
