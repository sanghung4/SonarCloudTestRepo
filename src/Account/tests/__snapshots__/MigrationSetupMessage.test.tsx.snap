// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Migration Setup Message should match snapshot in desktop 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1gcmr49-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-1xhj18k"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-xephrw-MuiTypography-root"
      >
        Welcome to
      </h3>
      <div
        class="MuiBox-root css-5wm1i5"
      >
        <svg
          height="25"
          width="75"
        >
          max.svg
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-1iq02cq-MuiGrid-root"
    >
      <svg>
        welcome-to-max.svg
      </svg>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-3qqvh9-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        We are excited to welcome you to our new and improved digital experience. Before we get started, please check your email for a link to reset your password.
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-14kxyr"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
        >
          Didn't get the email?
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1bl337n-MuiButtonBase-root-MuiButton-root"
        data-testid="resend-email-button"
        tabindex="0"
        type="button"
      >
        Resend Email
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1h16bbz-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-mkbj53"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-7lorw0-MuiTypography-root"
        >
          Having Trouble?
        </h4>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1hqugnb-MuiTypography-root"
        >
          We are here to help. 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1quwxlb-MuiTypography-root-MuiLink-root"
            to="/location-search"
          >
            Call a branch
          </a>
           to speak with a manager or 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1quwxlb-MuiTypography-root-MuiLink-root"
            to="/support"
          >
             contact eCommerce support.
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Migration Setup Message should match snapshot in desktop when location state is undefined 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1gcmr49-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-1xhj18k"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-xephrw-MuiTypography-root"
      >
        Welcome to
      </h3>
      <div
        class="MuiBox-root css-5wm1i5"
      >
        <svg
          height="25"
          width="75"
        >
          max.svg
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-1iq02cq-MuiGrid-root"
    >
      <svg>
        welcome-to-max.svg
      </svg>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-3qqvh9-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        We are excited to welcome you to our new and improved digital experience. Before we get started, please check your email for a link to reset your password.
      </p>
    </div>
  </div>
</div>
`;

exports[`Migration Setup Message should match snapshot in mobile 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1m24eu1-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-1xhj18k"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-xephrw-MuiTypography-root"
      >
        Welcome to
      </h3>
      <div
        class="MuiBox-root css-5wm1i5"
      >
        <svg
          height="25"
          width="75"
        >
          max.svg
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-yiprib-MuiGrid-root"
    >
      <svg>
        welcome-to-max.svg
      </svg>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-3kft1w-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
      >
        We are excited to welcome you to our new and improved digital experience. Before we get started, please check your email for a link to reset your password.
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-14kxyr"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1k41qqb-MuiTypography-root"
        >
          Didn't get the email?
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root  css-1bl337n-MuiButtonBase-root-MuiButton-root"
        data-testid="resend-email-button"
        tabindex="0"
        type="button"
      >
        Resend Email
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1h16bbz-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-mkbj53"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-7lorw0-MuiTypography-root"
        >
          Having Trouble?
        </h4>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1hqugnb-MuiTypography-root"
        >
          We are here to help. 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1quwxlb-MuiTypography-root-MuiLink-root"
            to="/location-search"
          >
            Call a branch
          </a>
           to speak with a manager or 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1quwxlb-MuiTypography-root-MuiLink-root"
            to="/support"
          >
             contact eCommerce support.
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
